BangazonWorkForceManagement/Controllers/DepartmentsController.cs

```diff
@@ -0,0 +1,241 @@
+                        }
+
+                        if (!reader.IsDBNull(reader.GetOrdinal("employeeId")))
+                        {
+                            int employeeId = reader.GetInt32(reader.GetOrdinal("employeeId"));
+                            if (!employeesSort.ContainsKey(employeeId))
+                            {
+                                Employee newEmployee = new Employee
+                                {
+                                    Id = employeeId,
+                                    FirstName = reader.GetString(reader.GetOrdinal("firstName")),
+                                    LastName = reader.GetString(reader.GetOrdinal("lastName")),
+                                    IsSuperVisor = false,
+                                    DepartmentId = reader.GetInt32(reader.GetOrdinal("departmentId"))
+                                };
+                                employeesSort.Add(employeeId, newEmployee);
+
+                                Department currentDepartment = departments[deptId];
+                                currentDepartment.Employees.Add(
+                                    new Employee
+                                    {
+                                        Id = employeeId,
+                                        FirstName = reader.GetString(reader.GetOrdinal("firstName")),
+                                        LastName = reader.GetString(reader.GetOrdinal("lastName")),
+                                        IsSuperVisor = false,
+                                        DepartmentId = reader.GetInt32(reader.GetOrdinal("departmentId"))
+                                    }
+                                    );
+                            }
+                        }
+                    }
+                    reader.Close();
+                    return View(departments.Values.ToList());
+                }
+            }
+        }
+
+        // GET: Departments/Details/5
+        public ActionResult Details(int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = $@"SELECT id, [name], budget FROM Department WHERE id = @id";
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    SqlDataReader reader = cmd.ExecuteReader();
+
+                    Department department = null;
+
+                    if (reader.Read())
+                    {
+                        department = new Department
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("id")),
+                            Name = reader.GetString(reader.GetOrdinal("name")),
+                            Budget = reader.GetInt32(reader.GetOrdinal("budget")),
+                            Employees = GetDeptEmployee(reader.GetInt32(reader.GetOrdinal("id")))
```

Comment:
I must have missed this one last time. Please use a join in your query instead of a separate query.

Reply:
my bad... it should be good now